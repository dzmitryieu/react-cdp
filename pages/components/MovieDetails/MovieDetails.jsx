import React from 'react';
import PropTypes from 'prop-types';
import styles from './MovieDetails.scss';
import defaultPoster from '../../../src/assets/images/default-poster.jpg';

const MovieDetails = (props) => {
  const {
    title,
    poster_path: posterPath,
    runtime,
    release_date: year,
    tagline,
    overview,
  } = props;
  const posterURL = posterPath || defaultPoster;
  return (
    <div className={styles.container}>
      <div className={styles.wrapper}>
        <div
          className={styles.poster}
          style={{
            backgroundImage: `url(${posterURL})`,
          }}
        />
        <div className={styles.description}>
          <h2 className={styles.title}>{title}</h2>
          <h4 className={styles.details}>{tagline}</h4>
          <div className={styles.statistics}>
            <p className={styles.year}>{year.split('-')[0]}</p>
            <p className={styles.duration}>
              {runtime && `${runtime} min`}
            </p>
          </div>
          <p className={styles.shortStory}>
            {overview}
          </p>
        </div>
      </div>
    </div>
  );
};

MovieDetails.propTypes = {
  title: PropTypes.string.isRequired,
  release_date: PropTypes.string.isRequired,
  runtime: PropTypes.number,
  tagline: PropTypes.string.isRequired,
  overview: PropTypes.string.isRequired,
  poster_path: PropTypes.string,
};
MovieDetails.defaultProps = {
  poster_path: null,
  runtime: null,
};

export default MovieDetails;
