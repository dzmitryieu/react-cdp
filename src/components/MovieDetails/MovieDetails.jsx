import React from 'react';
import PropTypes from 'prop-types';
import styles from './MovieDetails.scss';
import defaultPoster from '../../assets/images/default-poster.jpg';

const MovieDetails = (props) => {
  const {
    name,
    posterPath,
    duration,
    year,
    details,
    description,
  } = props;
  const posterURL = posterPath || defaultPoster;
  return (
    <div className={styles.container}>
      <div className={styles.wrapper}>
        <div
          className={styles.poster}
          style={{
            backgroundImage: `url(${posterURL})`,
          }}
        />
        <div className={styles.description}>
          <h2 className={styles.title}>{name}</h2>
          <h4 className={styles.details}>{details}</h4>
          <div className={styles.statistics}>
            <p className={styles.year}>{year}</p>
            <p className={styles.duration}>{duration}</p>
          </div>
          <p className={styles.shortStory}>
            {description}
          </p>
        </div>
      </div>
    </div>
  );
};

MovieDetails.propTypes = {
  name: PropTypes.string.isRequired,
  year: PropTypes.string.isRequired,
  duration: PropTypes.string.isRequired,
  details: PropTypes.string.isRequired,
  description: PropTypes.string.isRequired,
  posterPath: PropTypes.string,
};
MovieDetails.defaultProps = {
  posterPath: null,
};

export default MovieDetails;
