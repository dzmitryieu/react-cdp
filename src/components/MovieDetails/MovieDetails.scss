@import "../../utils/constants.scss";

.container {
  justify-content: center;
  display: flex;
  box-sizing: border-box;
  font-family: $helvetica;
  margin-top: 20px;
  width: 90%;
  align-self: center;

  .wrapper {
    height: 100%;
    width: 100%;
    position: relative;
    display: flex;
    flex-direction: row;
    justify-content: flex-start;

    .description {
      display: flex;
      flex-direction: column;
      justify-content: flex-start;

      .title {
        color: $light-red;
        font-size: 30px;
        margin-bottom: 5px;
      }

      .details {
        color: white;
        font-size: 14px;
        margin: 0;
      }

      .statistics {
        display: flex;
        justify-content: flex-start;
        align-items: center;
      }

      .year,
      .duration {
        font-size: 18px;
        color: white;
        font-weight: 600;
      }

      .duration {
        margin-left: 20px;
      }

      .shortStory {
        font-size: 18px;
        overflow: hidden;
        white-space: normal;
        color: white;
        text-align: left;
      }
    }

    .poster {
      background-size: cover;
      background-position: center center;
      background-repeat: no-repeat;
      height: 370px;
      min-width: 300px;
      text-align: center;
      margin: 0 50px 20px 20px;
    }
  }
}

@media screen and (max-width: $tablet-landscape-screen) {
  .container {
    width: 80%;
  }
}

@media screen and (max-width: $tablet-size-screen) {
  .container {
    width: 100%;

    .wrapper {
      .description {
        .shortStory {
          max-height: 100px;
          white-space: nowrap;
          text-overflow: ellipsis;
        }
      }
    }
  }
}

@media screen and (max-width: $phone-size-screen) {
  .container {
    position: relative;
    flex-direction: column;
    justify-content: center;

    .wrapper {
      position: relative;
      flex-direction: column;
      justify-content: center;

      .description {
        text-align: center;

        .statistics {
          justify-content: center;
        }
      }

      .poster {
        margin: 0 auto;
      }
    }
  }
}
