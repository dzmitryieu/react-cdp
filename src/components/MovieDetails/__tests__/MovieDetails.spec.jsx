import React from 'react';
import ShallowRenderer from 'react-test-renderer/shallow';
import MovieDetails from '../index';

const shallow = new ShallowRenderer();
const film = {
  name: 'Captain Marvel',
  genres: ['Drama', 'Comedy'],
  year: '1984',
  posterPath: '../../assets/images/poster.jpg',
  duration: '152 m',
  details: 'lorem lorem',
  description: 'lorem ipsum',
};
const film2 = {
  name: 'Captain Marvel',
  genres: ['Drama', 'Comedy'],
  year: '1984',
  duration: '152 m',
  details: 'lorem lorem',
  description: 'lorem ipsum',
};

describe('MoviDetails Snapshot', () => {
  test('render', () => {
    const component = shallow.render(
      <MovieDetails {...film} />,
    );
    expect(component).toMatchSnapshot();
  });
  test('render without PosterPath', () => {
    const component = shallow.render(
      <MovieDetails {...film2} />,
    );
    expect(component).toMatchSnapshot();
  });
});
