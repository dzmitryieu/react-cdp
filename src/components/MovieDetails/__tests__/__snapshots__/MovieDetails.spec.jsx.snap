// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MoviDetails Snapshot render 1`] = `
<div
  className="container"
>
  <div
    className="wrapper"
  >
    <div
      className="poster"
      style={
        Object {
          "backgroundImage": "url(../../assets/images/poster.jpg)",
        }
      }
    />
    <div
      className="description"
    >
      <h2
        className="title"
      >
        Captain Marvel
      </h2>
      <h4
        className="details"
      >
        lorem lorem
      </h4>
      <div
        className="statistics"
      >
        <p
          className="year"
        >
          1984
        </p>
        <p
          className="duration"
        >
          152 m
        </p>
      </div>
      <p
        className="shortStory"
      >
        lorem ipsum
      </p>
    </div>
  </div>
</div>
`;

exports[`MoviDetails Snapshot render without PosterPath 1`] = `
<div
  className="container"
>
  <div
    className="wrapper"
  >
    <div
      className="poster"
      style={
        Object {
          "backgroundImage": "url(test-file-stub)",
        }
      }
    />
    <div
      className="description"
    >
      <h2
        className="title"
      >
        Captain Marvel
      </h2>
      <h4
        className="details"
      >
        lorem lorem
      </h4>
      <div
        className="statistics"
      >
        <p
          className="year"
        >
          1984
        </p>
        <p
          className="duration"
        >
          152 m
        </p>
      </div>
      <p
        className="shortStory"
      >
        lorem ipsum
      </p>
    </div>
  </div>
</div>
`;
